libvast/vast/bloom_filter_synopsis.hpp

```diff
@@ -33,6 +33,14 @@ class bloom_filter_synopsis : public synopsis {
     // nop
   }
 
+  [[nodiscard]] synopsis_ptr clone() const override {
+    using self = bloom_filter_synopsis<T, HashFunction>;
+    // Cannot use `make_unique()` with a private constructor.
+    auto copy = std::unique_ptr<self>(new self(type()));
+    copy->bloom_filter_ = bloom_filter_;
+    return copy;
```

Comment:
Isn't this equivalent?
```suggestion
    return std::make_unique<std::decay_t<decltype(*this)>>(type(),
                                                           bloom_filter_);

```